Refactor launch_repairnator.sh

fermadeiral opened the pull request.

surli commented on repairnator/scripts/launch_dockerpool.sh:
```diff
@@ -0,0 +1,82 @@
+}
+
+if [ "$#" -eq 0 ]; then
+    echo "A list of build ids must be provided"
```
Put a usage at the beginning of the script. See what I did in `repair_buggy_build.sh`

fermadeiral replied:
I will do it.

surli commented on repairnator/scripts/launch_dockerpool.sh:
```diff
@@ -0,0 +1,82 @@
+    echo "$NB_LINES builds have been found."
+fi
+
+if [ "$#" -eq 1 ]; then
```
What's the purpose of this if? If `$# > 1` you don't execute those lines? There's a problem here.

fermadeiral replied:
> If $# > 1 you don't execute those lines?

Yes. It's because the `launch_dockerpool.sh` (and also `launch_scanner.sh`) can be executed from `launch_repairnator.sh`. So, if this is the case, the first stuff to do, like to init and create the structure, should be done only once (i.e. in `launch_repairnator.sh`), thus those lines can't be executed. See in `launch_repairnator.sh` how I call `launch_dockerpool.sh` and `launch_scanner.sh`. There is an extra arg to control what I just described. I found no way better to do that. If you have any suggestion, please, let me know.

surli replied:
In that case, 
  1. We should be calling only init_scripts.sh which should contain also content of create_structure.sh
  2. When init_script.sh is called, it should initialize a variable like "REPAIRNATOR_INITIALIZED" and you should test the value of this variable.

fermadeiral replied:
Perfect.

surli commented on repairnator/scripts/launch_dockerpool.sh:
```diff
@@ -0,0 +1,82 @@
+  fi
+}
+
+if [ "$#" -eq 0 ]; then
```
If there should be exactly one argument, maybe use `-ne 1` instead of `-eq 0`

fermadeiral replied:
Actually `launch_dockerpool.sh` needs to receive one arg (the list of builds), but it can also receive one more arg if it's running throught `launch_repairnator.sh`.

surli replied:
> it can also receive one more arg

Which one? I cannot see it in `launch_repairnator.sh`

fermadeiral replied:
Now `launch_dockerpool.sh` receives only one arg, the list of builds. The other one was to control if the things had been initialized, but with your approach we don't need it anymore.

surli replied:
So we can use `-ne 1` here...

fermadeiral replied:
Done.

surli commented on repairnator/scripts/launch_scanner.sh:
```diff
@@ -16,16 +16,18 @@ function ca {
 }
 
-SCRIPT_DIR="$(dirname "${BASH_SOURCE[0]}")"
+if [ "$#" -eq 0 ]; then
```
I don't understand this if...

fermadeiral replied:
`launch_scanner.sh` doesn't require any arg. But if this is running throught `launch_repairnator.sh`, it can't execute those lines because the things were already initialized.

surli replied:
See my answer above. Current design is really unclear.

fermadeiral replied:
I agree.

surli commented:
Have you tested the scripts before I merge?

fermadeiral commented:
> Have you tested the scripts before I merge?

I tested more the ones which were the center of the refactoring: `launch_repairnator.sh`, `launch_scanner.sh` and `launch_dockerpool.sh`. They seem to work properly. The others I just run until the main process starts (to see if the initialization is ok, which is what I changed on them). I had an issue with `check_branches.sh` but I just fixed it. They also seem to work properly.

surli commented:
OK don't bother too much with check_branches.sh in the future: it's really deprecated, it needs to be refactored.
